msal/src/Microsoft.Identity.Client/AuthorizationUserAgent.cs

```diff
@@ -0,0 +1,50 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Possible values for the MSAL Authorization User Agent.
+    /// </summary>
+    public enum AuthorizationUserAgent

```

Comment:
I need to get more clarity than is in Shane's spec for what this is used for...

Reply:
I suppose this is the flip we have today in our mobile setups (ios and android) where we allow both the system browser and the web view. Most likely these will also come to other platforms in time event hough on some platforms the system browser experience is less than ideal from a usability perspective. Also for .netcore we will likely see the system browser first and then potentially in time also the embedded webview.

---
In reply to: [237701361](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1442#discussion_r237701361) [](ancestors = 237701361)